### PR TITLE
Feature: add useCardLayoutFrom prop for Tables

### PR DESCRIPTION
**Pull request by baomastr**
https://app.clickup.com/t/2gvefed

### CODE REVIEW

File: src/components/v2/Table/Head.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ function Head<C extends { key: string; label: string; orderable: boolean }[]>({
   onRequestOrder,
   className,
 }: IHeadProps<C>) {
-  const styles = useStyles();
+  const styles = useStyles({});
```
Comment (therealemjy) [inline src/components/v2/Table/Head.tsx]: I'd make the param optional instead. Passing an empty object doesn't make sense

File: src/components/v2/Table/index.tsx [hunk 2]
```diff
@@ -97,17 +100,17 @@ export const Table = ({
...
+  }, [columns, gridTemplateColumns, isCardLayout]);
 
   return (
-    <Paper css={[styles.root, isSmDown && styles.rootMobile]} className={className}>
-      {title && <h4 css={[styles.title, isSmDown && styles.titleMobile]}>{title}</h4>}
+    <Paper css={[styles.root, isCardLayout && styles.rootMobile]} className={className}>
+      {title && <h4 css={[styles.title, isCardLayout && styles.titleMobile]}>{title}</h4>}
 
-      {isSmDown ? (
+      {isCardLayout ? (
```
Comment (therealemjy) [inline src/components/v2/Table/index.tsx:113]: As we spoke about earlier this week, we want to move away from using Javascript and use CSS instead. You can render both variants and use a CSS media query to hide one or the other using `display: none`.

File: src/components/v2/Table/index.tsx [hunk 3]
```diff
@@ -7,8 +7,9 @@ import TableContainer from '@mui/material/TableContainer';
 import TableRow from '@mui/material/TableRow';
 import Paper from '@mui/material/Paper';
 import Typography from '@mui/material/Typography';
+import { Breakpoint } from '@mui/material';
```
Comment (therealemjy) [inline src/components/v2/Table/index.tsx]: This should use the properties of our `BREAKPOINT` constant from the theme instead

**Review by therealemjy — CHANGES_REQUESTED** [review]

File: src/components/v2/Table/index.tsx (hunk 2 above)
Comment (baomastr) [reply src/components/v2/Table/index.tsx:113]: i think, hook use is justified here.
we have different layouts for this cases. that's such hooks are made for.

File: src/components/v2/Table/index.tsx (hunk 3 above)
Comment (baomastr) [reply src/components/v2/Table/index.tsx]: done

File: src/components/v2/Table/Head.tsx (hunk 1 above)
Comment (baomastr) [reply src/components/v2/Table/Head.tsx]: done

**Pull request closed without merging**